### PR TITLE
soxr: expose more resampler's settings to the end user

### PR DESCRIPTION
**Pull request by godsic**
At the moment Audacious' soxr plug-in exposes limited subset of the re-sampler's setting to end user. This commit address this problem by exposing the following settings:
- re-sampler's phase response (minimum phase, intermediate phase and linear phase);
- one can now opt for the steep filter;
- one can allow/forbid aliasing;
- one can opt for ultra high re-sampling quality (aka 32 bit).

### CODE REVIEW

File: src/soxr/sox-resampler.cc [hunk 1]
```diff
@@ -96,7 +96,14 @@ void SoXResampler::start (int & channels, int & rate)
     if (new_rate == rate)
         return;
 
-    soxr_quality_spec_t q = soxr_quality_spec (aud_get_int ("soxr", "quality"), 0);
+    int quality = aud_get_int ("soxr", "quality");
+    int phase_response = aud_get_int ("soxr", "phase_response_list");
+    int use_steep_filter = (aud_get_bool ("soxr", "use_steep_filter")) ? SOXR_STEEP_FILTER : 0;
+    int allow_aliasing = (aud_get_bool ("soxr", "allow_aliasing")) ? SOXR_ALLOW_ALIASING : 0;
+
+    unsigned long quality_recipe = quality | phase_response | use_steep_filter | allow_aliasing;
```
Comment (jlindgren90) [inline src/soxr/sox-resampler.cc]: You don't need an unsigned long to OR int's together.

File: src/soxr/sox-resampler.cc [hunk 2]
```diff
@@ -152,13 +159,25 @@ static const ComboItem method_list[] = {
...
     ComboItem (N_("Medium"), SOXR_MQ),
     ComboItem (N_("High"), SOXR_HQ),
-    ComboItem (N_("Very High"), SOXR_VHQ)
+    ComboItem (N_("Very High"), SOXR_VHQ),
+    ComboItem (N_("Ultra High"), SOXR_32_BITQ)
+};
+
+static const ComboItem phase_response_list[] = {
+    ComboItem (N_("Minimum"), SOXR_MINIMUM_PHASE),
+    ComboItem (N_("Intermidiate"), SOXR_INTERMEDIATE_PHASE),
```
Comment (jlindgren90) [inline src/soxr/sox-resampler.cc]: Spelling: Intermediate

File: src/soxr/sox-resampler.cc [hunk 3]
```diff
@@ -96,7 +96,14 @@ void SoXResampler::start (int & channels, int & rate)
     if (new_rate == rate)
         return;
 
-    soxr_quality_spec_t q = soxr_quality_spec (aud_get_int ("soxr", "quality"), 0);
+    int quality = aud_get_int ("soxr", "quality");
+    int phase_response = aud_get_int ("soxr", "phase_response_list");
```
Comment (jlindgren90) [inline src/soxr/sox-resampler.cc]: Please add defaults for the new settings to the SoXResampler::defaults table earlier in the file.  Use "TRUE" or "FALSE" for Booleans, and aud::numeric_string for enumeration values.

**Comment by godsic** [comment]
@jlindgren90 thanks for the valuable comments. I have just pushed the requested changes.

**Pull request merged**